satellite/gracefulexit/endpoint.go

```diff
@@ -169,6 +179,10 @@ func (endpoint *Endpoint) doProcess(stream processStream) (err error) {
 		if err != nil {
```

Comment:
That could be an issue. I made a PR to address this concern: https://github.com/storj/storj/pull/3357

Reply:
I think in the worst case scenario, if a node connects to `doProcess` in parallel, the satellite would likely send multiple piece transfer requests for the same pieces to the node. Which means that we might end up with duplicate pieces/order limits across multiple "recipient" nodes.
But on the satellite side, we should be able to handle this so that only the first transfer is actually successful, and the others are ignored. And the duplicate pieces would eventually be picked up by garbage collection.
In other words, it is not ideal, but I don't think it is necessarily a breaking issue. What do you think?